### DISCUSSION TITLE
Swift Intellisense

### DISCUSSION THREAD

**Discussion by nadavmoreno** [zed-industries/zed#23615] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

While clicking on certain code elements, I want it to "Go to Definition" and open new tabs with that definition if needed

### If applicable, add mockups / screenshots to help present your vision of the feature

Using Zed, while hovering over `View`, I'm getting this valuable information:

<img width="855" alt="image" src="https://github.com/zed-industries/zed/assets/22380885/978610e9-6ae4-439c-9c41-ea51fb323611">

I want to be able to click on `View`, like I'm able to do on Xcode:
<img width="689" alt="image" src="https://github.com/zed-industries/zed/assets/22380885/f03592a9-9385-4f35-a47e-2a7a9a2865ea">

**Comment by romans-ovo** [comment]
Is Cmd+Click not working?

**Comment by th0jensen** [comment]
I have no idea what the difference is between this integration with VS Code and the LSP that Zed uses, but Apple announced this at WWDC: <https://github.com/swiftlang/vscode-swift>

It seems to add a bunch of new features to 3rd party editors, and it would be cool to see this integrated into Zed, in some sort of way, if possible.

(cmd-click go to definition is included in this VSCode extension)

**Comment by davidzech27** [comment]
Unfortunately SourceKit-LSP doesn't support Xcode projects out of the box, so you may be interested in this: https://github.com/SolaWing/xcode-build-server
